chia/full_node/mempool.py

```diff
@@ -73,7 +86,8 @@ def remove_from_pool(self, items: List[bytes32]) -> None:
             self.total_mempool_cost -= item.cost
             assert self.total_mempool_cost >= 0
             mempool_info = self.get_mempool_info()
-            self.fee_estimator.remove_mempool_item(mempool_info, item)
+            if reason != MempoolRemoveReason.BLOCK_INCLUSION:
```

Comment:
this looks like a change in behavior. would you mind explaining it?

Reply:
Yes. This is the first in a series of small fixes that fix broken behaviour in the fee estimator, specifically how and where the code is called from.

This particular line fixes the issue that MempoolItems were being marked as removed in the fee estimator even though the reason they were being removed was that they had successfully gotten into a block.

Reply:
I could take that line out and put it in the PR that does have a user-visible fix.

Let me know what you think.

Reply:
Yeah, I think I'll do that.